component/file_cache/file_cache.go

```diff
@@ -780,7 +780,7 @@ func (fc *FileCache) OpenFile(options internal.OpenFileOptions) (*handlemap.Hand
 		}
 
 		// Open the file in write mode.
-		f, err = os.OpenFile(localPath, os.O_CREATE|os.O_WRONLY, options.Mode)
+		f, err = os.OpenFile(localPath, os.O_CREATE|os.O_RDWR, options.Mode) // need these permissions for file share to work
```

Comment:
@vibhansa-msft Is this safe? The file share SDK expects Read access to the file as well in the WriteToFile method

Reply:
This is ok, when we are downloading the file we can keep this permissions. Anyway this handle will be closed as soon as download is complete and then we will open another handle with permissions user requested.